src/main/java/com/google/sps/servlets/ShowInterviewersServlet.java

```diff
@@ -0,0 +1,137 @@
+import com.google.sps.data.ScheduledInterview;
+import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.lang.Integer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletException;
+import javax.servlet.RequestDispatcher;
+import java.util.Optional;
+
+@WebServlet("/show-interviewers")
+public class ShowInterviewersServlet extends HttpServlet {
+
+  private AvailabilityDao availabilityDao;
+  private PersonDao personDao;
+
+  @Override
+  public void init() {
+    init(new DatastoreAvailabilityDao(), new DatastorePersonDao());
+  }
+
+  public void init(AvailabilityDao availabilityDao, PersonDao personDao) {
+    this.availabilityDao = availabilityDao;
+    this.personDao = personDao;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String utc = request.getParameter("utc");
+    TimeRange range =
+        new TimeRange(Instant.parse(utc), Instant.parse(utc).plus(1, ChronoUnit.HOURS));
+    List<Availability> availabilitiesInRange =
+        availabilityDao.getInRangeForAll(range.start(), range.end());
+    List<Person> possiblePeople = getPossiblePeople(availabilitiesInRange, range);
+    Set<PossibleInterviewer> possibleInterviewers = peopleToPossibleInterviewers(possiblePeople);
+    request.setAttribute("interviewers", possibleInterviewers);
+    RequestDispatcher rd = request.getRequestDispatcher("/possibleInterviewers.jsp");
+    try {
+      rd.forward(request, response);
+    } catch (ServletException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  List<Person> getPossiblePeople(List<Availability> allAvailabilities, TimeRange range) {
+    Set<String> allInterviewers = new HashSet<String>();
+    for (Availability avail : allAvailabilities) {
+      allInterviewers.add(avail.userId());
+    }
+
+    // We don't want to schedule an interview for a user with themself, so we are removing
+    // the current user's id from the list.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
+    // replaced with this hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
```

Comment:
Remove spaces to keep the logical chunk together (this chunk being related to dropping the user)

Reply:
Done